### PR TITLE
Write unit tests for components, actions, and reducers

### PR DESCRIPTION
**Pull request by arinaldi**

### CODE REVIEW

File: src/__tests__/AsyncActionCreators.test.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  RECEIVE_LOCATION,
+  CHANGE_STATUS,
+} from '../constants';
+const fetchMock = require('fetch-mock');
+
+const middleware = [thunk];
+const mockStore = configureMockStore(middleware);
+
+describe('async actions', () => {
+  it('successfully fetches a location', () => {
```
Comment (ashleymichal) [inline src/__tests__/AsyncActionCreators.test.js]: this isn't what you are testing.

File: src/__tests__/CurrentLocation.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+const location = {
+  date: 'May 2',
+  name: 'Richmond',
+  aqi: 99,
+  description: 'Great air quality',
+  color: 'green',
+};
+
+function setup() {
```
Comment (ashleymichal) [inline src/__tests__/CurrentLocation.test.js:13]: wai?

File: src/__tests__/HistoryTableRow.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    props,
+    enzymeWrapper,
+  };
+}
+
+describe('HistoryTableRow component', () => {
+  it('renders self', () => {
+    const { enzymeWrapper, props } = setup();
+
+    expect(enzymeWrapper.find('td')).toHaveLength(5);
```
Comment (ashleymichal) [inline src/__tests__/HistoryTableRow.test.js]: generally avoid multiple "expects" in one "it". look at Jest's snapshot feature if what you are trying to capture is a consistent render behavior.

File: src/__tests__/LocalStorage.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    color: 'red',
+  },
+];
+
+describe('LocalStorage', () => {
+  it('saves last 5 searches to localStorage', () => {
+    locations.forEach((location) => {
+      saveToHistory(location);
+    });
+    expect(getHistory()).toHaveLength(5);
```
Comment (ashleymichal) [inline src/__tests__/LocalStorage.test.js:47]: what if I try to save another location?

File: src/__tests__/Reducers.test.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    );
+  });
+
+  it('handles RECEIVE_LOCATION', () => {
+    const date = new Date();
+    const name = 'Richmond, VA';
+    const aqi = 62;
+    const description = 'Fair air quality';
+    const color = 'rgb(181, 226, 30)';
+    const expectedAction = {
```
Comment (ashleymichal) [inline src/__tests__/Reducers.test.js]: the action isn't part of your expectation, the state of your store after you call that action on your reducer is your expectation.

File: src/actions/index.js [hunk 6]
```diff
@@ -31,10 +31,11 @@ export function fetchLocation(location) {
 
     return geocodeGoogle(location)
       .then((data) => {
-        if (data.status === 'OK') {
+        if (data.status === 'ZERO_RESULTS') {
+          dispatch(changeStatus('Invalid location', 'error'));
+        } else if (data.status === 'OK') {
           return data;
         }
```
Comment (ashleymichal) [inline src/actions/index.js]: are there other possible `data.status` messages returned by this API? You may consider a switch statement with a default case for unknown statuses.

File: src/api/index.js [hunk 7]
```diff
@@ -11,21 +11,26 @@ export function geocodeGoogle(location) {
 
   return fetch(url)
     .then((res) => {
-      if (res.ok) return res.json();
-      throw new Error('Bad network response');
+      // if (!res.ok) throw new Error('TONY TEST');
```
Comment (ashleymichal) [inline src/api/index.js]: remove commented code from commits.

File: src/api/index.js [hunk 8]
```diff
@@ -11,21 +11,26 @@ export function geocodeGoogle(location) {
...
+      return res.json();
     })
     .then((json) => {
-      const name = json.results[0].formatted_address;
-      const lat = json.results[0].geometry.location.lat;
-      const lng = json.results[0].geometry.location.lng;
+      const { results, status } = json;
+      if (results.length !== 0) {
```
Comment (ashleymichal) [inline src/api/index.js]: `undefined !== 0 => true`
`undefined  >  0 => false`

**Review by ashleymichal — CHANGES_REQUESTED** [review]

File: src/api/index.js (hunk 7 above)
Comment (arinaldi) [reply src/api/index.js]: Done.

File: src/__tests__/LocalStorage.test.js (hunk 4 above)
Comment (arinaldi) [reply src/__tests__/LocalStorage.test.js:47]: Done.

File: src/__tests__/HistoryTableRow.test.js (hunk 3 above)
Comment (arinaldi) [reply src/__tests__/HistoryTableRow.test.js]: Done.

File: src/__tests__/CurrentLocation.test.js (hunk 2 above)
Comment (arinaldi) [reply src/__tests__/CurrentLocation.test.js:13]: I found this in the React Components example on the Redux testing page:

http://redux.js.org/docs/recipes/WritingTests.html#components

File: src/api/index.js (hunk 8 above)
Comment (arinaldi) [reply src/api/index.js]: Done.

File: src/actions/index.js (hunk 6 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/__tests__/Reducers.test.js (hunk 5 above)
Comment (arinaldi) [reply src/__tests__/Reducers.test.js]: Done.

File: src/__tests__/AsyncActionCreators.test.js (hunk 1 above)
Comment (arinaldi) [reply src/__tests__/AsyncActionCreators.test.js]: Done.

**Pull request merged**